src/fu-main.c

```diff
@@ -1539,6 +1584,15 @@ main (int argc, char *argv[])
 	g_signal_connect (priv->engine, "percentage-changed",
 			  G_CALLBACK (fu_main_engine_percentage_changed_cb),
 			  priv);
+
+	if (!fu_engine_load_config (priv->engine, FU_ENGINE_LOAD_FLAG_NONE, &error))
+		return EXIT_FAILURE;
```

Comment:
Why do we have to use `fu_engine_load_config()` before parsing the arguments?

Reply:
Otherwise all the initialization is missed. The argument parsing is when debugging gets turned on.